src/AppInstallerCLICore/Argument.h

```diff
@@ -36,6 +36,38 @@ namespace AppInstaller::CLI
         Flag,
     };
 
+    // Categories an arg type can belong to.
+    // Used to reason about the arguments present without having to repeat the same
+    // lists every time.
+    enum class ArgTypeCategory

```

Comment:
>ArgTypeCategory [](http://example.com/codeflow?start=15&length=15)

Do we want to add the "CopyInSubContext" to the category here too? The idea you previously had for copying args in subcontext.